src/hotspot/share/opto/loopnode.cpp

```diff
@@ -4138,7 +4138,7 @@ void PhaseIdealLoop::collect_useful_template_assertion_predicates(Unique_Node_Li
 
 void PhaseIdealLoop::collect_useful_template_assertion_predicates_for_loop(IdealLoopTree* loop,
                                                                            Unique_Node_List &useful_predicates) {
-  Node* entry = loop->_head->in(LoopNode::EntryControl);
+  Node* entry = loop->_head->as_Loop()->skip_strip_mined()->in(LoopNode::EntryControl);
```

Comment:
looks like a bug-fix?

Reply:
Is now required due to changing `can_apply_loop_predication()` which is checked before calling this method.